### DISCUSSION TITLE
Graph from pandas dataframe

### DISCUSSION THREAD

**Discussion by davidshumway** [RDFLib/rdflib#1666] [Documentation]
Does rdflib support loading a graph from a pandas dataframe. For example, networkx has this functionality, as described here: (https://networkx.org/documentation/networkx-1.10/reference/generated/networkx.convert_matrix.from_pandas_dataframe.html)

**Comment by ghost** [comment]
Not directly. But a pandas dataframe can be [exported as csv](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html) and read back in with the RDFLib [command-line tool cvs2rdf](https://github.com/RDFLib/rdflib/blob/master/rdflib/tools/csv2rdf.py).

If the data is in HDF5 format, pandas can export dataframes as HDF5 and there's an [HDF5-to-RDF reader](https://github.com/HDFGroup/h5ld) available.
